File: android/app/src/main/kotlin/com/tbtrapp/components/ContactRow.kt
package com.tbtrapp.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tbtrapp.data.AppContact

@Composable
fun ContactRow(
    contact: AppContact,
    onClick: () -> Unit,
    onAvatarClick: (() -> Unit)? = null
) {
    Card(
        shape = RoundedCornerShape(16.dp),
        elevation = CardDefaults.cardElevation(4.dp),
        modifier = Modifier
            .fillMaxWidth()
            .clickable { onClick() }
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(14.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            CachedProfileAvatar(
                uid = contact.uid,
                name = contact.name,
                size = 44.dp,
                onClick = onAvatarClick
            )

            Spacer(modifier = Modifier.width(14.dp))

            Column(modifier = Modifier.weight(1f)) {
                Text(text = contact.name, fontWeight = FontWeight.SemiBold, fontSize = 15.sp)
                Text(text = contact.phone, fontSize = 12.sp, color = Color.Gray)
            }
        }
    }
}

File: lib/screens/components/contact_row.dart
import 'package:flutter/material.dart';
import '../../data/app_contact.dart';
import 'profile_avatar.dart';

class ContactRow extends StatelessWidget {
  final AppContact contact;
  final VoidCallback onTap;
  final VoidCallback? onAvatarTap;

  const ContactRow({
    super.key,
    required this.contact,
    required this.onTap,
    this.onAvatarTap,
  });

  @override
  Widget build(BuildContext context) {
    return Card(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      elevation: 4,
      child: InkWell(
        borderRadius: BorderRadius.circular(16),
        onTap: onTap,
        child: Padding(
          padding: const EdgeInsets.all(14),
          child: Row(
            children: [
              CachedProfileAvatar(uid: contact.uid, name: contact.name, size: 44, onTap: onAvatarTap),
              const SizedBox(width: 14),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(contact.name, style: const TextStyle(fontWeight: FontWeight.w600, fontSize: 15)),
                    Text(contact.phone, style: const TextStyle(fontSize: 12, color: Colors.grey)),
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}